launcher/ui/themes/ThemeManager.cpp

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: GPL-3.0-only
+/*
+ *  Prism Launcher
+ *  Copyright (C) 2022 Tayou <tayou@gmx.net>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "ThemeManager.h"
+
+#include "ui/themes/SystemTheme.h"
+#include "ui/themes/DarkTheme.h"
+#include "ui/themes/BrightTheme.h"
+#include "ui/themes/CustomTheme.h"
+
+#ifdef Q_OS_WIN
+#include "ui/WinDarkmode.h"
+#endif
+
+ThemeManager::ThemeManager(MainWindow* mainWindow) {
+    m_mainWindow = mainWindow;

```

Comment:
btw, you can use the initializer list :)
for a pointer it won't change a lot though, but still good practice.
```suggestion
ThemeManager::ThemeManager(MainWindow* mainWindow) : m_mainWindow{ mainWindow }
{
```

Reply:
i did actually see that before, in other places in code and wanted to use it, but couldn't find it anymore when I was working on that part. thank you!

Reply:
you can see that here :)
https://en.cppreference.com/w/cpp/language/constructor

Reply:
you can see the example with comments: https://en.cppreference.com/w/cpp/language/constructor#Example